lms/djangoapps/courseware/courses.py

```diff
@@ -415,3 +416,32 @@ def get_studio_url(course, page):
     if is_studio_course and is_mongo_course:
         studio_link = get_cms_course_link(course, page)
     return studio_link
+
+
+def get_entrance_exam_modules(request, course, student):
+    """
+    This returns all problems in an entrance exam
+    """
+
+    exam_modules = []
+    if settings.FEATURES.get('ENTRANCE_EXAMS', False):
+        if getattr(course, 'entrance_exam_enabled', False):
+            exam_key = UsageKey.from_string(course.entrance_exam_id)
+            exam_descriptor = modulestore().get_item(exam_key)
+
+            def create_module(descriptor):
+                """creates an XModule instance given a descriptor"""
+                field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+                    course.id, student, descriptor
+                )
+                return get_module_for_descriptor(student, request, descriptor, field_data_cache, course.id)
+
+            exam_modules = yield_dynamic_descriptor_descendents(
+                exam_descriptor,
+                create_module
+            )
+            ignore_categories = ['course', 'chapter', 'sequential', 'vertical']
+            # filter not graded modules and anything which is not a problem
+            exam_modules = [module for module in exam_modules
+                            if module.graded and module.category not in ignore_categories]
```

Comment:
Just a note here -- this particular change (if needed) should be made in the parent branch